test/util/cert_generation.go

```diff
@@ -15,6 +15,32 @@ import (
 	"github.com/spiffe/spire/test/clock"
 )
 
+// NewCSRTemplate returns a default CSR template with the specified SPIFFE ID.
+func NewCSRTemplate(spiffeId string) ([]byte, error) {
```

Comment:
Consider having this return the public key. This would enable you to ensure that the certificate returned in the SubmitCSR response is valid for the submitted CSR.

Reply:
I fixied with https://github.com/spiffe/spire/pull/988/commits/60b87ed52f580b8fa16b896623799b1dfde76f03 .

Reply:
I made a mistake and returned the private key instead of the public key.
I fixed it with https://github.com/spiffe/spire/pull/988/commits/067fcbe63ee011cde6a2cb4c2d404bf0b6d9ed7a .